### ISSUE TITLE
Missing "installed" statuses in the app

### ISSUE THREAD

**Issue by artginzburg** [GetOvert/Overt#39]
I have a lot of casks installed via brew, though Overt only shows 3 of them as installed. Interestingly, the statuses are missing for `homebrew/cask` (Official tap), but are present for `artginzburg/tap` (Unofficial tap).

Here's my `brew list`:
```
==> Casks
adoptopenjdk	discord		gamemaker	middle		piposs
appcleaner	fig		github		middleclick	postman
blender		firefox		google-chrome	opera-gx	telegram
darkmode	folx		keybase		overt		zoom
defold		functiontoggler	maccy		parallels
```

Here's the GUI part that I expect to have all of this casks:
<img width="992" alt="Screenshot 2023-01-25 at 02 43 13" src="https://user-images.githubusercontent.com/46137336/214437561-289acf2c-3a74-4de2-86d8-6c413c1315b4.png">

Searching for the casks individually also does not help — they appear as "not installed" (did not expect it to help, but wanted to double-check).

**Comment by ThatsJustCheesy** [comment]
If you don't mind, could you post the output of

```
brew info --json=v2 --cask --installed
```

as well as the contents of Overt's dev console? (Settings → Catalog → "Rebuild Now", then "View" menu → Toggle Developer Tools → Console tab → Right-click → Save as…)

For performance reasons, installed official casks are actually indexed twice—once by simply fetching from Homebrew's JSON API, and once by running the command above, which provides details about what's installed locally. I think something is going wrong during this second step.

**Comment by artginzburg** [comment]
`brew info --json=v2 --cask --installed`: [brewInfo.log](https://github.com/GetOvert/Overt/files/10530431/brewInfo.log)
The contents of Overt's dev console after selecting Filter "Installed": [-1675014029992.log](https://github.com/GetOvert/Overt/files/10530433/-1675014029992.log)

**Comment by ThatsJustCheesy** [comment]
Do you have `HOMEBREW_INSTALL_FROM_API` enabled?

The casks from `homebrew/cask` have `tap: null` in the JSON output instead of `tap: "homebrew/cask"`. From a cursory look, it seems this is generated by (at least) the ["install from API" loader](https://github.com/Homebrew/brew/blob/b4510401bdbb0a1291519555875b93dabca75b2c/Library/Homebrew/cask/cask_loader.rb#L225) (no `tap:` argument)

**Comment by artginzburg** [comment]
Hmm. I don't remember setting any additional configuration for Homebrew, or installing casks in any other way than just `brew install --cask`. But I recently reinstalled brew (keeping all the installed casks/formulae). Maybe this is what has reset the tap to `null`.

I'm now reading some `brew info` logs. What do you think about relying on the `"installed"` property for this filter? It's `null` if not installed, and a version string (`"2.7.4"`) otherwise.

**Comment by ThatsJustCheesy** [comment]
OK, interesting. The fix I'll use for now is to just default the tap to `homebrew/cask` if it's `null`.

And that `"installed"` property is indeed checked :) it's just that the mitigation for #25 (https://github.com/GetOvert/Overt/commit/d8cdc52c8f30f406907fa5145757a8e6e75d9bf5) was to skip packages that are missing required fields, and `tap` is one of those required fields, so the casks with `"tap": null` are ignored.

**Comment by artginzburg** [comment]
Also wanted to suggest this fix, but I don’t know what happens then if something was installed using a path to a local formula

**Comment by ThatsJustCheesy** [comment]
Does this build work for you? (Manually reindex with Settings → Catalog → "Rebuild Now")

https://1drv.ms/u/s!Av3LlU_IAIhlxt4noVB4tF8S2vAkTg?e=NwS52a

**Comment by artginzburg** [comment]
Yep, this works: <img width="1624" alt="Screenshot 2023-01-29 at 23 47 49" src="https://user-images.githubusercontent.com/46137336/215352032-36a198dd-4b28-469f-9165-ae2420de2d61.png">

**Comment by ThatsJustCheesy** [comment]
Fix is present in [release 0.5.6](https://github.com/GetOvert/Overt/releases/tag/v0.5.6).
